model/src/w3iopomd.F90

```diff
@@ -1025,8 +1025,471 @@ SUBROUTINE W3IOPE ( A )
+         '              ARRAY DIMENSIONS : ',2I8/)
+    !
+1000 FORMAT (/' *** WAVEWATCH III ERROR IN W3IOPO : '/               &
+         '     ERROR IN OPENING FILE'/                          &
+         '     IOSTAT =',I5/)
+1001 FORMAT (/' *** WAVEWATCH III ERROR IN W3IOPO : '/               &
+         '     PREMATURE END OF FILE'/)
+1002 FORMAT (/' *** WAVEWATCH III ERROR IN W3IOPO : '/               &
+         '     ERROR IN READING FROM FILE'/                     &
+         '     IOSTAT =',I5/)
+    !
+#ifdef W3_T
+9000 FORMAT (' TEST W3IOPO : IPASS =',I4,'    INXOUT = ',A,       &
+         ' WRITE = ',L1,' UNIT =',I3/                         &
+         '               IGRD =',I3,' FEXT = ',A)
+
+9001 FORMAT (' TEST W3IOPO : OPENING NEW FILE [',A,']')
+9002 FORMAT (' TEST W3IOPO : TEST PARAMETERS:'/                   &
+         '       IDSTR : ',A/                                 &
+         '      VEROPT : ',A/                                 &
+         '      NK,NTH :',I5,I8/                              &
+         '        NOPT :',I5)
+9003 FORMAT (' TEST W3IOPO : POINT LOCATION AND ID')
+9004 FORMAT (3X,I4,2F10.2,2X,A)
+    !
+9010 FORMAT (' TEST W3IOPO : TIME  :',I9.8,I7.6)
+9011 FORMAT (' TEST W3IOPO : END OF FILE REACHED')
+    !
+9020 FORMAT (' TEST W3IOPO : POINT NR.:',I5)
+9021 FORMAT (' TEST W3IOPO :',2I4,2F6.3)
+9022 FORMAT (' TEST W3IOPO :',4I7,2X,4I2,2X,4F5.2)
+9030 FORMAT (' TEST W3IOPO :',F8.1,2(F7.2,F7.1))
+#endif
+    !/
+    !/ End of W3IOPON ----------------------------------------------------- /
+    !/
+    
+  END SUBROUTINE W3IOPON
+  !/ ------------------------------------------------------------------- /
+  !>
   !> @brief Read/write point output.
   !>
+  !> The format of the point output file is:
+  !> Size (bytes) | Type | Variable | Meaning
+  !> -------------|------|----------|--------
+  !> 40 | character*40 | IDTST | ID string
+  !> 4 | integer | VERTST | version of test?
+  !> 4 | integer | NK | ???
+  !> 4 | integer | MTH | ???
+  !> 4 | integer | NOPTS | ???
+  !> 8*NOPTS | integer(2,NOPTS) | PTLOC | Point locations
+  !> 7*NOPTS | character*7 | PTNME | Point names
+  !> 8 | integer(2) | TIME | Time
+  !> reclen*NOPTS | * | * | record
+
+  !> Each record contains:
+  !> Size (bytes) | Type | Variable | Meaning
+  !> -------------|------|----------|--------
+  !> 4 | integer | IW | ???
+  !> 4 | integer | II | ???
+  !> 4 | integer | IL | ???
+  !> 4 | integer | DPO | ???
+  !> 4 | integer | WAO | ???
+  !> 4 | integer | WDO | ???
+  !> 4 | integer | TAUAO | (W3_FLX5 only) ???
+  !> 4 | integer | TAIDO | (W3_FLX5 only) ???
+  !> 4 | integer | DAIRO | (W3_FLX5 only) ???
+  !> 4 | integer | ZET_SETO | (W3_SETUP only) ???
+  !> 4 | integer | ASO | ???
+  !> 4 | integer | CAO | ???
+  !> 4 | integer | CDO | ???
+  !> 4 | integer | ICEO | ???
+  !> 4 | integer | ICEHO  | ???
+  !> 4 | integer | ICEFO | ???
+  !> 4 | integer | GRDID | ???
+  !> 4 | integer | SPCO(J,I),J=1,NSPEC | ???
```

Comment:
@MatthewMasarik-NOAA and @JessicaMeixner-NOAA I am documenting the format of the binary points file in doxygen.

I have listed all the variables read from the file. But what are they and what do they mean? I need your help to fill in the answers...

Reply:
Hi @edwardhartnett, I can help with that.  Let's talk today when we meet.